SEBrowser/Controllers/OpenXDAController.cs

```diff
@@ -219,32 +225,47 @@ private string getEventTypeFilter(EventSearchPostData postData)
 
         private string getPhaseFilter(EventSearchPostData postData)
         {
-            List<string> phase = new List<string>();
-
-            if (postData.PhaseA && postData.PhaseB && postData.PhaseC)
-                return "";
-            if (!postData.PhaseA && !postData.PhaseB && !postData.PhaseC)
+            Dictionary<string, bool> phases = new Dictionary<string, bool> 
+            {
+                ["AN"] = postData.PhaseAN,
+                ["BN"] = postData.PhaseBN,
+                ["CN"] = postData.PhaseCN,
+                ["AB"] = postData.PhaseAB,
+                ["BC"] = postData.PhaseBC,
+                ["ABG"] = postData.PhaseABG,
+                ["BCG"] = postData.PhaseBCG,
+                ["ABC"] = postData.PhaseABC,
+                ["ABCG"] = postData.PhaseABCG
+            };
+
+            bool allTrue = true;
+            bool allFalse = true;
+            foreach (var value in phases.Values) 
+            {
+                if (value)
+                {
+                    allFalse = false;
+                }
+                else
+                {
+                    allTrue = false;
+                }
+            }
+            if (allTrue || allFalse)
+            {
                 return "";
-
-            if (postData.PhaseA)
-            {
-                phase.Add("(SELECT Phase.Name FROM Disturbance LEFT JOIN Phase ON Disturbance.PhaseID = Phase.ID WHERE Disturbance.ID = EventWorstDisturbance.WorstDisturbanceID) IN ('AN','AB','CA')");
-                phase.Add("FaultSummary.FaultType IN ('AN','AB','CA', 'ABG', 'CAG', 'ABC', 'ABCG')");
-            }
-            if (postData.PhaseB)
-            {
-                phase.Add("(SELECT Phase.Name FROM Disturbance LEFT JOIN Phase ON Disturbance.PhaseID = Phase.ID WHERE Disturbance.ID = EventWorstDisturbance.WorstDisturbanceID) IN ('BN','AB','BC')");
-                phase.Add("FaultSummary.FaultType IN ('BN','AB','BC', 'ABG', 'BCG', 'ABC', 'ABCG')");
-            }
-            if (postData.PhaseC)
-            {
-                phase.Add("(SELECT Phase.Name FROM Disturbance LEFT JOIN Phase ON Disturbance.PhaseID = Phase.ID WHERE Disturbance.ID = EventWorstDisturbance.WorstDisturbanceID) IN ('CN','BC','CA')");
-                phase.Add("FaultSummary.FaultType IN ('CN','BC','CA', 'BCG', 'CAG', 'ABC', 'ABCG')");
             }
 
-
-
-            return string.Join(" OR ", phase);
+            string phaseCombined = "";
+            foreach (var item in phases) 
+            {
+                if (item.Value) 
+                {
+                    phaseCombined += ("\'" + item.Key + "\', ");
+                }
+            }
+            phaseCombined = phaseCombined.Substring(0, phaseCombined.Length - 2);
+            return $"(Event.ID in (SELECT Disturbance.EventID FROM Disturbance WHERE Disturbance.PhaseID IN (SELECT Phase.ID FROM Phase WHERE Phase.Name IN ({phaseCombined}))) OR FaultSummary.FaultType IN ({phaseCombined}))";
```

Comment:
I don't think this is right. @gcsantos-gpa or @StephenCWills can you look at this?

I believe the whole statement is effectively a UNION of 3 different cases. If an event is `Transient, Sag, Interuption or Swell it will take the `Disturbance` records, if it is a Fault it will take the `FaultSummary` record. 
In the first case there may be multiple `Disturbance.EventID` with different Phases, In the second and there should only be one. If an Event is not in `FaultSummary` or `Disturbance` it won't have a Phase (I think) so that case can be excluded. For case 1 you would need to check `Disturbance.PhaseID` for case 2 you should check `FaultSummary.FaultType` since both of these are `NULL` in the other case the 2 checks would need to be `OR`ed